app/utils/IncidentP5Helper.scala

```diff
@@ -66,17 +70,43 @@ class IncidentP5Helper(
     call = None
   )
 
+  def unLocodeRow: Option[SummaryListRow] = buildRowFromAnswer[String](
+    answer = data.Location.UNLocode,
+    formatAnswer = formatAsText,
+    prefix = "departure.notification.incident.index.unLocode",
+    id = Some(s"unLocode-$displayIndex"),
+    call = None
+  )
+
+  def addressRow: Option[SummaryListRow] = buildRowFromAnswer[String](
+    answer = Some("address"), // TODO: Pull from incident data,
+    formatAnswer = formatAsText,
+    prefix = "departure.notification.incident.index.address",
+    id = Some(s"address-$displayIndex"),
+    call = None
+  )
+
   def incidentInformationSection: StaticSection = StaticSection(
     sectionTitle = None,
     rows = Seq(
       incidentCodeRow,
       descriptionRow,
       countryRow,
       identifierTypeRow,
-      coordinatesRow
+      selectLocationRow(data.Location.qualifierOfIdentification)
     ).flatten
   )
 
+  private val selectLocationRow: String => Option[SummaryListRow] = qualifierOfIdentification =>
+    qualifierOfIdentification.toUpperCase match {
+      case "W" => coordinatesRow
+      case "U" => unLocodeRow
+      case "Z" => addressRow
+      case _ =>
+        logger.error(s"Unexpected qualifier of identification: $qualifierOfIdentification")
+        None
+    }
```

Comment:
Strictly speaking we're not really bothered about this logic. If one is defined the other two will be undefined. We can just safely add `coordinatesRow`, `unLocodeRow` and `addressRow` to our list of rows

Reply:
Yea just to hop on this, I'd had a quick look before I went on lunch - I was of the some opinion as Jack that we should be able to have them all in the list or rows as theoretically only one should ever be defined. I think the confusion has probably come from the IE182 postman data as more than one of these has been populated but in reality this shouldn't happen